src/lib/utils.ts

```diff
@@ -14,6 +14,14 @@ export function getNanoTimestamp() {
 
 export const pseudoRandomBytesAsync = promisify(pseudoRandomBytes);
 
+export const isValidUrl = (url: string) => {
+	try {
+		return Boolean(new URL(url));
```

Comment:
Relying on `new URL` doesn't necessary mean the input is a valid URL per se since it will do some coercion as according to https://nodejs.org/docs/latest-v14.x/api/url.html#url_new_url_input_base where it will attempt to coerce into a valid string, eg `{ toString: () => 'https://example.org/' }` would be valid despite not being a URL string and it will convert unicode characters into ascii using punycode, but those characters from the original string aren't strictly valid (they rely on whatever is using them being aware of how to convert them to ascii before querying dns). There's also the question of if we want to restrict to certain protocols eg https

Reply:
@Page- since we control the contracts repo I don't think we need a very strict check, but I agree that this function doesn't take care of any edge-cases. What do you think, do we keep it like this or shall I go with a more elaborate check using some package or regex?

Reply:
I thought the idea was that we could theoretically have other contracts repos supplied? In which case I'd like to have decent checks, especially since I can easily see this function being reused by other non-contracts code.
There is `https://www.npmjs.com/package/validator` which we use elsewhere to validate emails that has an `isURL` and might be an improvement, I'm not sure how it compares though

Reply:
We won't introduce new contract repos, but people could PR to our contracts repo (which we need to review), but I agree, let's go with a more robust check.

Reply:
We don't intend to introduce new contract repos today, but I could definitely imagine it in future, whether for balenahub or community supported device types or whatever so I'd prefer to play it safe. It also protects against the code being reused or copy pasted somewhere where it really does matter - I've definitely seen a pattern that is ok in one place reused in a place it's not ok before